README.md

```diff
@@ -1,8 +1,107 @@
 # ECM Distro Tools
 
-This repo ECM Distro Tools (codename: dirt-weasel) is a collection of programs, scripts, and utilities that provide for easier administration, management, and interaction with the RKE2 and K3s ecosystems.
+ECM Distro Tools (codename: dirt-weasel) is a collection of programs, scripts, and utilities that provide for easier administration, management, and interaction with the RKE2 and K3s ecosystems.
 
 ## Building
+There's a mix of Go and shell scripts in this repository. The shell scripts reside in the `bin` directory and are ready to use. The Go programs are rooted in the `cmd` directory and need to be compiled. To compile the Go programs run the following in the root of the project:
+```sh
+make all
+```
+To compile the container image locally, perform:
+
+```sh
+docker build . -t rancher/ecm-distro-tools:local
+```
+## Utility Index 
+The following is a detailed list of the utilities included in this repository and their corresponding usage.
+
+### update_go
+
+```sh
```

Comment:
We can probably get away with showing an example or 2 and leave off the entire help menu for each script / app in the repo.

Reply:
sounds good

Reply:
e.g.

```sh
docker run --rm -i -t --env GITHUB_TOKEN=<TOKEN> rancher/ecm-distro-tools weekly_report -r k3s-io/k3s
```